### PR TITLE
Make function/cursor keys work. Right now ignores keypad mode and alt…

### PR DESCRIPTION
**Pull request by durgadas311**
…ernate ESC codes.

It works, at least for MMS system utilities. Right now it ignores any terminal settings like keypad mode and alternate ESC codes. I had a problem with the two-char sequences causing overrun in the serial device, so I added a usleep(2000) (2mS) between. I did not try other values or work out an alternate scheme. With no flow-control it's difficult.

Mapped F1-F5, cursor and HOME, and F6->BLUE, F7->RED, F->GRAY.

### CODE REVIEW

File: Src/main.cpp [hunk 1]
```diff
@@ -161,20 +161,56 @@ void special(int key, int x, int y)
...
+        h89.keypress('R' | 0x80);
+        break;
+
+    case GLUT_KEY_HOME:
+        h89.keypress('H' | 0x80);
+        break;
+
     case GLUT_KEY_UP:
-        h89.keypress('8');
+        h89.keypress('A' | 0x80);
```
Comment (mgarlanger) [inline Src/main.cpp]: Can you add a 'TODO' to fully handle all variations of shifted and alt keypad values.

Comment (durgadas311) [inline Src/main.cpp]: I updated the commit (forced update) with some comments. See if those better describe the situation.

Comment (mgarlanger) [inline Src/main.cpp]: What I was thinking. On a modern keyboard even thought there is both the arrow keys and the number pad, the number pad will send arrows,pageUp/Down, etc, if the num lock is turned off. It also has an 'alternate mode', which sends different ESC sequences. See page 22 in the H19 manual(http://koyado.com/Heathkit/H8-Z19_files/Heath%20Z19%20Terminal%20Operating%20Manual.pdf) - there are four potential states. The eventual plan, is to have they numpad send the right values based on the current shifted and alternate modes set by ESC sequences and/or dipswitch settings on the terminal board.

**Comment by durgadas311** [comment]
Added a new commit to master, expected to be able to create a separate pull request but it seems not.

This new commit re-orgs the GUI code, I think like you were talking about. All the GLUT code is pulled into H19. and main now allows selection of what "Console" object to use. I made an example StdioConsole (activated by commandline option "-g stdio") that is like a dump TTY. Not perfect, but a good proof-of-concept. I also created "Console" as a type of "Terminal" object that allows for an out-of-band communication for sending commands to do things like replace a floppy disk image (remove/insert diskette). Still more work to go that direction, but it's a framework.

File: Src/Console.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/// \file Console.cpp
+///
+/// \date Feb 6, 2016
+///
+/// \author Douglas Miller
+///
+
+#include "Console.h"
+
+Console::Console(int argc, char **argv)
```
Comment (mgarlanger) [inline Src/Console.cpp:10]: What's the plan for argc & argv?

File: Src/h19.cpp [hunk 3]
```diff
@@ -3,12 +3,36 @@
...
+#include "config.h"
+#include <iostream>
+#include <signal.h>
+#include <stdlib.h>
+
+#if OGL
+#ifdef __APPLE__
+#include <OpenGL/gl.h>
+#include <OpenGL/glu.h>
+#include <GLUT/glut.h>
```
Comment (mgarlanger) [inline Src/h19.cpp:15]: duplicate includes, glut.h was already included.

File: Src/Console.cpp (hunk 2 above)
Comment (durgadas311) [inline Src/Console.cpp:10]: I was trying to make things more flexible. My thought was to be able to 
customize certain types of console, for example if we make a 
SocketConsole we could get IP address and port (to listen on) off the 
commandline.

File: Src/h19.cpp (hunk 3 above)
Comment (durgadas311) [inline Src/h19.cpp:15]: I fixed this in master-next.

Fixed other problems you pointed out on that branch, too.

**Pull request merged**